pkg/blobstore/cloud/cloud_blob_access.go

```diff
@@ -76,13 +80,58 @@ func (ba *cloudBlobAccess) Put(ctx context.Context, digest digest.Digest, b buff
 }
 
 func (ba *cloudBlobAccess) FindMissing(ctx context.Context, digests digest.Set) (digest.Set, error) {
+	// Workers send their errors here
```

Comment:
Does this actually use HTTP pipelining, or are we spawning O(n) goroutines to hammer S3?

I think a change like this needs more evaluation before we can follow it through. I guess it's wiser to just focus on the BeforeCopyFunc parts for now.

Reply:
Very much the latter - but splitting it out is reasonable, I'll turn it into a separate PR.

(Note that we're not transferring any data, and the transport layer is delegated to the aws sdk, but I would expect the transport layer to do the right thing - spawning lots of goroutines is the normal API here)